src/main/resources/avro/sequenceAnnotations.avdl

```diff
@@ -0,0 +1,227 @@
+@namespace("org.ga4gh.models")
+/**
+This protocol defines annotations on GA4GH genomic sequences It includes two
+types of annotations: continuous and discrete hierarchical.
+
+The discrete hierarchical annotations are derived from the Sequence Ontology
+(SO) and GFF3 work 
+
+   http://www.sequenceontology.org/gff3.shtml
+
+The goal is to be able to store annotations using the GFF3 and SO conceptual
+model, although there is not necessarly a one-to-one mapping in Avro records
+to GFF3 records.
+
+The minimum requirement is to be able to accurately represent the current
+state of the art annotation data and the full SO model.  Feature is the
+core generic record which corresponds to the a GFF3 record.
+*/
+protocol SequenceAnnotations {
+
+  import idl "common.avdl";
+  import idl "ontologies.avdl";
+
+  /**
+  Identifier from a public database
+  */
+  record ExternalIdentifier {
+    /**
+    The source of the identifier.
+    (e.g. `Ensembl`)
+    */
+    string database;
+
+    /**
+    The ID defined by the external database.
+    (e.g. `ENST00000000000`)
+    */
+    string identifier;
+
+    /**
+    The version of the object or the database
+    (e.g. `78`)
+    */
+    string version;
+  }
+
+  /**
+  Type defining a collection of attributes associated with various protocol
+  records.  Each attribute is a name that maps to an array of one or more
+  values.  Values can be strings, external identifiers, or ontology terms.
+  Values should be split into the array elements instead of using a separator
+  syntax that needs to parsed.
+  */
+  // TODO: how are multiple instances of a given attribute vs multiple values
+  // for an attribute distinguished
+  record Attributes {
+    map<array<union {string, ExternalIdentifier, OntologyTerm}>> vals = {};
```

Comment:
OntologyTerm needs to be brought in from the metadata branch.
For a brief moment, it was in a separate file, but should be in
metadata.advl now.



Maciek Smuga-Otto <notifications@github.com> writes:
> In src/main/resources/avro/sequenceAnnotations.avdl:
> 
> > +    (e.g. `78`)
> > +    */
> > +    string version;
> > +  }
> > +
> > +  /**
> > +  Type defining a collection of attributes associated with various protocol
> > +  records.  Each attribute is a name that maps to an array of one or more
> > +  values.  Values can be strings, external identifiers, or ontology terms.
> > +  Values should be split into the array elements instead of using a separator
> > +  syntax that needs to parsed.
> > +  */
> > +  // TODO: how are multiple instances of a given attribute vs multiple values
> > +  // for an attribute distinguished
> > +  record Attributes {
> > +    map<array<union {string, ExternalIdentifier, OntologyTerm}>> vals = {};
> 
> OntologyTerm is missing from the API (see similar comment elsewhere for more)
> 
> —
> Reply to this email directly or view it on GitHub.*
>